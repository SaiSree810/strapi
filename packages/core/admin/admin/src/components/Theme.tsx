import * as React from 'react';

import { DesignSystemProvider } from '@strapi/design-system';
import { useIntl } from 'react-intl';
import { createGlobalStyle } from 'styled-components';

import { useThemeToggle } from '../../hooks/useThemeToggle';

interface ThemeProps {
  children: React.ReactNode;
}

const Theme = ({ children }: ThemeProps) => {
  const { currentTheme, themes } = useThemeToggle();
  const { locale } = useIntl();

  return (
    <DesignSystemProvider locale={locale} theme={themes[currentTheme] || themes.light}>
      {children}
      <GlobalStyle />
    </DesignSystemProvider>
  );
};

const GlobalStyle = createGlobalStyle`
  body {
    background: ${({ theme }) => theme.colors.neutral100};
  }
`;

export { Theme };
