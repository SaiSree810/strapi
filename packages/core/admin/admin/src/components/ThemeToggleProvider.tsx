import * as React from 'react';

import { DefaultTheme } from 'styled-components';

import { ThemeToggleContext, ThemeName } from '../contexts/themeToggle';

const THEME_KEY = 'STRAPI_THEME';

const getDefaultTheme = () => {
  const persistedTheme = localStorage.getItem(THEME_KEY) as ThemeName | null;

  return persistedTheme || 'system';
};

interface ThemeToggleProviderProps {
  children: React.ReactNode;
  themes: {
    light: DefaultTheme;
    dark: DefaultTheme;
  };
}

const ThemeToggleProvider = ({ children, themes }: ThemeToggleProviderProps) => {
  const [currentTheme, setCurrentTheme] = React.useState<ThemeName>(getDefaultTheme());

  const handleChangeTheme = React.useCallback(
    (nextTheme: ThemeName) => {
      setCurrentTheme(nextTheme);
      localStorage.setItem(THEME_KEY, nextTheme);
    },
    [setCurrentTheme]
  );

  const themeValues = React.useMemo(() => {
    return {
      currentTheme,
      onChangeTheme: handleChangeTheme,
      themes,
    };
  }, [currentTheme, handleChangeTheme, themes]);

  return <ThemeToggleContext.Provider value={themeValues}>{children}</ThemeToggleContext.Provider>;
};

export { ThemeToggleProvider };
export type { ThemeToggleProviderProps };
