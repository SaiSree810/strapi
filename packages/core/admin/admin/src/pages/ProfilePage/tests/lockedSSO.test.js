import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { QueryClient, QueryClientProvider } from 'react-query';
import { lightTheme, darkTheme } from '@strapi/design-system';
import ProfilePage from '../index';
import serverLockedSSO from './utils/serverLockedSSO';
import ThemeToggleProvider from '../../../components/ThemeToggleProvider';
import Theme from '../../../components/Theme';

jest.mock('../../../components/LocalesProvider/useLocalesProvider', () => () => ({
  changeLocale() {},
  localeNames: ['en'],
  messages: ['test'],
}));

jest.mock('@strapi/helper-plugin', () => ({
  ...jest.requireActual('@strapi/helper-plugin'),
  useNotification: jest.fn(),
  useFocusWhenNavigate: jest.fn(),
  useAppInfo: jest.fn(() => ({ setUserDisplayName: jest.fn() })),
  useOverlayBlocker: jest.fn(() => ({ lockApp: jest.fn, unlockApp: jest.fn() })),
}));

const client = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

const App = (
  <QueryClientProvider client={client}>
    <IntlProvider messages={{}} textComponent="span" locale="en">
      <ThemeToggleProvider themes={{ light: lightTheme, dark: darkTheme }}>
        <Theme>
          <ProfilePage />
        </Theme>
      </ThemeToggleProvider>
    </IntlProvider>
  </QueryClientProvider>
);

describe('ADMIN | Pages | Profile page', () => {
  beforeAll(() => serverLockedSSO.listen());

  beforeEach(() => {
    jest.clearAllMocks();
  });

  afterEach(() => {
    serverLockedSSO.resetHandlers();
  });

  afterAll(() => {
    jest.resetAllMocks();
    serverLockedSSO.close();
  });

  it('renders and matches the snapshot', async () => {
    const { container } = render(App);
    await waitFor(() => {
      expect(screen.getByText('Interface language')).toBeInTheDocument();
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should display username if it exists', async () => {
    render(App);
    await waitFor(() => {
      expect(screen.getByText('yolo')).toBeInTheDocument();
    });
  });

  it('should not display the change password section if the user role is Locked', async () => {
    render(App);
    await waitFor(() => {
      expect(screen.queryByText('Change password')).toBeNull();
    });
  });
});
