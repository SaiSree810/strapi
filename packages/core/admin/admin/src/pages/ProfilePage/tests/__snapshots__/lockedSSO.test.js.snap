// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADMIN | Pages | Profile page renders and matches the snapshot 1`] = `
.c1 {
  background: #f6f6f9;
  padding-top: 40px;
  padding-right: 56px;
  padding-bottom: 40px;
  padding-left: 56px;
}

.c3 {
  min-width: 0;
}

.c13 {
  padding-right: 56px;
  padding-left: 56px;
}

.c7 {
  background: #4945ff;
  padding: 8px;
  padding-right: 16px;
  padding-left: 16px;
  border-radius: 4px;
  border-color: #4945ff;
  border: 1px solid #4945ff;
  cursor: pointer;
}

.c12 {
  padding-bottom: 56px;
}

.c15 {
  background: #ffffff;
  padding-top: 24px;
  padding-right: 32px;
  padding-bottom: 24px;
  padding-left: 32px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c29 {
  background: #ffffff;
  padding-right: 12px;
  padding-left: 12px;
  border-radius: 4px;
  position: relative;
  overflow: hidden;
  width: 100%;
  cursor: default;
}

.c32 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c37 {
  background: transparent;
  border-radius: 4px;
  cursor: pointer;
}

.c6 {
  font-weight: 600;
  font-size: 2rem;
  line-height: 1.25;
  color: #32324d;
}

.c11 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #ffffff;
}

.c17 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 1.25;
  color: #32324d;
}

.c21 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #32324d;
}

.c23 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #d02b20;
}

.c27 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #32324d;
}

.c34 {
  font-size: 0.875rem;
  line-height: 1.43;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #32324d;
}

.c40 {
  font-size: 0.75rem;
  line-height: 1.33;
  color: #666687;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c14 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 24px;
}

.c16 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
}

.c20 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
}

.c30 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 16px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c33 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 12px;
}

.c9 {
  position: relative;
  outline: none;
}

.c9 > svg {
  height: 12px;
  width: 12px;
}

.c9 > svg > g,
.c9 > svg path {
  fill: #ffffff;
}

.c9[aria-disabled='true'] {
  pointer-events: none;
}

.c9:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c9:focus-visible {
  outline: none;
}

.c9:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c10 {
  height: 2rem;
}

.c10[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c10[aria-disabled='true'] .c5 {
  color: #666687;
}

.c10[aria-disabled='true'] svg > g,.c10[aria-disabled='true'] svg path {
  fill: #666687;
}

.c10[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c10[aria-disabled='true']:active .c5 {
  color: #666687;
}

.c10[aria-disabled='true']:active svg > g,.c10[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c10:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c10:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c10 svg > g,
.c10 svg path {
  fill: #ffffff;
}

.c22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c24 {
  line-height: 0;
}

.c26 {
  border: none;
  border-radius: 4px;
  padding-bottom: 0.65625rem;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 0.65625rem;
  color: #32324d;
  font-weight: 400;
  font-size: 0.875rem;
  display: block;
  width: 100%;
  background: inherit;
}

.c26::-webkit-input-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c26::-moz-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c26:-ms-input-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c26::placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c26[aria-disabled='true'] {
  color: inherit;
}

.c26:focus {
  outline: none;
  box-shadow: none;
}

.c25 {
  border: 1px solid #dcdce4;
  border-radius: 4px;
  background: #ffffff;
  outline: none;
  box-shadow: 0;
  -webkit-transition-property: border-color,box-shadow,fill;
  transition-property: border-color,box-shadow,fill;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
}

.c25:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c38 {
  border: none;
}

.c38 svg {
  height: 0.6875rem;
  width: 0.6875rem;
}

.c38 svg path {
  fill: #666687;
}

.c31 {
  border: 1px solid #dcdce4;
  min-height: 2.5rem;
  outline: none;
  box-shadow: 0;
  -webkit-transition-property: border-color,box-shadow,fill;
  transition-property: border-color,box-shadow,fill;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
}

.c31[aria-disabled='true'] {
  color: #666687;
}

.c31:focus-visible {
  outline: none;
}

.c31:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c39 > svg {
  width: 0.375rem;
}

.c39 > svg > path {
  fill: #666687;
}

.c35 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c36 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c41[data-state='checked'] .c5 {
  font-weight: bold;
  color: #4945ff;
}

.c18 {
  display: grid;
  grid-template-columns: repeat(12,1fr);
  gap: 20px;
}

.c19 {
  grid-column: span 6;
  max-width: 100%;
}

.c0:focus-visible {
  outline: none;
}

.c28 {
  color: #4945ff;
}

@media (max-width:68.75rem) {
  .c19 {
    grid-column: span 12;
  }
}

@media (max-width:34.375rem) {
  .c19 {
    grid-column: span;
  }
}

<main
  aria-busy="false"
  aria-labelledby="main-content-title"
  class="c0"
  id="main-content"
  tabindex="-1"
>
  <form
    action="#"
    novalidate=""
  >
    <div
      style="height: 0px;"
    >
      <div
        class="c1"
        data-strapi-header="true"
      >
        <div
          class="c2"
        >
          <div
            class="c3 c4"
          >
            <h1
              class="c5 c6"
            >
              yolo
            </h1>
          </div>
          <button
            aria-disabled="false"
            class="c7 c8 c9 c10"
            type="submit"
          >
            <div
              aria-hidden="true"
              class=""
            >
              <svg
                fill="none"
                height="1rem"
                viewBox="0 0 24 24"
                width="1rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20.727 2.97a.2.2 0 0 1 .286 0l2.85 2.89a.2.2 0 0 1 0 .28L9.554 20.854a.2.2 0 0 1-.285 0l-9.13-9.243a.2.2 0 0 1 0-.281l2.85-2.892a.2.2 0 0 1 .284 0l6.14 6.209L20.726 2.97Z"
                  fill="#212134"
                />
              </svg>
            </div>
            <span
              class="c5 c11"
            >
              Save
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="c12"
    >
      <div
        class="c13"
      >
        <div
          class="c14"
        >
          <div
            class="c15"
          >
            <div
              class="c16"
            >
              <h2
                class="c5 c17"
              >
                Profile
              </h2>
              <div
                class="c18"
              >
                <div
                  class="c19"
                >
                  <div
                    class=""
                  >
                    <div>
                      <div
                        class=""
                      >
                        <div
                          class="c20"
                        >
                          <label
                            class="c5 c21 c22"
                            for="firstname"
                          >
                            First name
                            <span
                              class="c5 c23 c24"
                            >
                              *
                            </span>
                          </label>
                          <div
                            class="c2 c25"
                          >
                            <input
                              aria-disabled="false"
                              aria-invalid="false"
                              aria-required="true"
                              class="c26"
                              id="firstname"
                              name="firstname"
                              placeholder=""
                              type="text"
                              value="michoko"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="c19"
                >
                  <div
                    class=""
                  >
                    <div>
                      <div
                        class=""
                      >
                        <div
                          class="c20"
                        >
                          <label
                            class="c5 c21 c22"
                            for="lastname"
                          >
                            Last name
                          </label>
                          <div
                            class="c2 c25"
                          >
                            <input
                              aria-disabled="false"
                              aria-invalid="false"
                              aria-required="false"
                              class="c26"
                              id="lastname"
                              name="lastname"
                              placeholder=""
                              type="text"
                              value="ronronscelestes"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="c19"
                >
                  <div
                    class=""
                  >
                    <div>
                      <div
                        class=""
                      >
                        <div
                          class="c20"
                        >
                          <label
                            class="c5 c21 c22"
                            for="email"
                          >
                            Email
                            <span
                              class="c5 c23 c24"
                            >
                              *
                            </span>
                          </label>
                          <div
                            class="c2 c25"
                          >
                            <input
                              aria-disabled="false"
                              aria-invalid="false"
                              aria-required="true"
                              class="c26"
                              id="email"
                              name="email"
                              placeholder=""
                              type="email"
                              value="michka@michka.fr"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="c19"
                >
                  <div
                    class=""
                  >
                    <div>
                      <div
                        class=""
                      >
                        <div
                          class="c20"
                        >
                          <label
                            class="c5 c21 c22"
                            for="username"
                          >
                            Username
                          </label>
                          <div
                            class="c2 c25"
                          >
                            <input
                              aria-disabled="false"
                              aria-invalid="false"
                              aria-required="false"
                              class="c26"
                              id="username"
                              name="username"
                              placeholder=""
                              type="text"
                              value="yolo"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c15"
          >
            <div
              class="c16"
            >
              <div
                class="c20"
              >
                <h2
                  class="c5 c17"
                >
                  Experience
                </h2>
                <span
                  class="c5 c27"
                >
                  Preference changes will apply only to you. More information is available 
                  <a
                    class="c28"
                    href="https://docs.strapi.io/developer-docs/latest/development/admin-customization.html#locales"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    here
                  </a>
                  .
                </span>
              </div>
              <div
                class="c18"
              >
                <div
                  class="c19"
                >
                  <div
                    class=""
                  >
                    <div
                      class=""
                    >
                      <div
                        class="c20"
                      >
                        <label
                          class="c5 c21 c22"
                          for=":r8:"
                        >
                          Interface language
                        </label>
                        <div
                          aria-autocomplete="none"
                          aria-controls="radix-:rb:"
                          aria-describedby=":r8:-hint :r8:-error"
                          aria-expanded="false"
                          aria-label="Interface language"
                          class="c29 c30 c31"
                          data-state="closed"
                          dir="ltr"
                          id=":r8:"
                          overflow="hidden"
                          role="combobox"
                          tabindex="0"
                        >
                          <span
                            class="c32 c33"
                          >
                            <span
                              class="c5 c34 c35"
                            >
                              <span
                                class="c36"
                              />
                            </span>
                          </span>
                          <span
                            class="c33"
                          >
                            <button
                              aria-label="Clear the interface language selected"
                              class="c37 c38"
                              role="button"
                              tabindex="0"
                              title="Clear the interface language selected"
                            >
                              <svg
                                fill="none"
                                height="1rem"
                                viewBox="0 0 24 24"
                                width="1rem"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M24 2.417 21.583 0 12 9.583 2.417 0 0 2.417 9.583 12 0 21.583 2.417 24 12 14.417 21.583 24 24 21.583 14.417 12 24 2.417Z"
                                  fill="#212134"
                                />
                              </svg>
                            </button>
                            <span
                              aria-hidden="true"
                              class="c39"
                            >
                              <svg
                                fill="none"
                                height="1rem"
                                viewBox="0 0 14 8"
                                width="1rem"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  clip-rule="evenodd"
                                  d="M14 .889a.86.86 0 0 1-.26.625L7.615 7.736A.834.834 0 0 1 7 8a.834.834 0 0 1-.615-.264L.26 1.514A.861.861 0 0 1 0 .889c0-.24.087-.45.26-.625A.834.834 0 0 1 .875 0h12.25c.237 0 .442.088.615.264a.86.86 0 0 1 .26.625Z"
                                  fill="#32324D"
                                  fill-rule="evenodd"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                        <p
                          class="c5 c40"
                          id=":r8:-hint"
                        >
                          This will only display your own interface in the chosen language.
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="c19"
                >
                  <div
                    class=""
                  >
                    <div
                      class=""
                    >
                      <div
                        class="c20"
                      >
                        <label
                          class="c5 c21 c22"
                          for=":rc:"
                        >
                          Interface mode
                        </label>
                        <div
                          aria-autocomplete="none"
                          aria-controls="radix-:rf:"
                          aria-describedby=":rc:-hint :rc:-error"
                          aria-expanded="false"
                          aria-label="Interface mode"
                          class="c29 c30 c31"
                          data-state="closed"
                          dir="ltr"
                          id=":rc:"
                          overflow="hidden"
                          role="combobox"
                          tabindex="0"
                        >
                          <span
                            class="c32 c33"
                          >
                            <span
                              class="c5 c34 c35"
                            >
                              <span
                                class="c36"
                              >
                                Light mode
                              </span>
                            </span>
                          </span>
                          <span
                            class="c33"
                          >
                            <span
                              aria-hidden="true"
                              class="c39"
                            >
                              <svg
                                fill="none"
                                height="1rem"
                                viewBox="0 0 14 8"
                                width="1rem"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  clip-rule="evenodd"
                                  d="M14 .889a.86.86 0 0 1-.26.625L7.615 7.736A.834.834 0 0 1 7 8a.834.834 0 0 1-.615-.264L.26 1.514A.861.861 0 0 1 0 .889c0-.24.087-.45.26-.625A.834.834 0 0 1 .875 0h12.25c.237 0 .442.088.615.264a.86.86 0 0 1 .26.625Z"
                                  fill="#32324D"
                                  fill-rule="evenodd"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                        <p
                          class="c5 c40"
                          id=":rc:-hint"
                        >
                          Displays your interface in the chosen mode.
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</main>
`;
